Makefile

```diff
@@ -29,3 +31,10 @@ bdd-test: clean
 .PHONY: clean
 clean:
 	rm -f coverage.txt
+	rm -Rf ./build
+
+.PHONY: generate-openapi-spec
+generate-openapi-spec: clean
+	@echo "Generating and validating controller API specifications using Open API"
```

Comment:
I wonder why the command that generates the spec file needs to validate it as well.

Reply:
swagger doesn't provide two-in-one command which generates & validates in one go.
`swagger generate` can fail if there is a syntax or missing fields in annotations.
`swagger validate` fails if any request/response model is missing or if there are any orphan models or if any misconfiguration under 'Open API Specification 2.0' standards.

Reply:
I was imprecise - by "command" I meant the `make` recipe "generate-openapi-spec".

My assumption is that if `swagger generate` succeeds then `swagger validate` will succeed as well. If `swagger generate` fails then `make generate-openapi-spec` fails and `swagger validate` won't be executed at all.

Reply:
I understand what you are saying, but your assumption is wrong here.

`swagger generate` passes 90% of the time since it fills spec with defaults if any information is missing. It fails if there any compilation error and if there any annotation syntax issue.

`swagger validate` fails most of the time since it never scans code, it scans generated spec and make sure all necessary info and all mappings are intact (between path, models etc), and there aren't any orphan request/response/models.

Reply:
Oh ok - I think I see now. So, for example, there is the possibility that `swagger generate` will output orphaned models and then `swagger validate` would detect this error. Did I get it right?

Reply:
Yes, and generate scans code, validate scans spec.